### PR TITLE
MS Graph Handle 404 Response

### PR DESCRIPTION
**Pull request by darkushin**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/34324

## Description
Added 404 error handling to the MicrosoftApiModule.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests

### CODE REVIEW

File: Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py [hunk 1]
```diff
@@ -127,6 +128,14 @@ def http_request(
         if is_response_empty_and_successful and return_empty_response:
             return response
 
+        # Handle 404 errors instead of raising them as exceptions:
+        if response.status_code == 404:
+            try:
+                error_message = response.json()
+            except Exception:
+                error_message = 'Page Not Found - 404 Response'
```
Comment (yuvalbenshalom) [inline Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py:136]:
```suggestion
                error_message = 'Not Found - 404 Response'
```
in this case it is not a page

File: Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py [hunk 2]
```diff
@@ -531,3 +540,14 @@ def device_auth_request(self) -> str:
             return_error(f'Error in Microsoft authorization: {str(e)}')
         set_integration_context({'device_code': response_json.get('device_code')})
         return response_json.get('user_code', '')
+
+
+class NotFoundError(Exception):
+    """Exception raised for 404 - Page Not Found errors.
```
Comment (yuvalbenshalom) [inline Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py:546]:
```suggestion
    """Exception raised for 404 - Not Found errors.
```

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
looks good. see my comments.

**Comment by darkushin** [comment]
@amshamah419 Please review the release notes. Please review only [RN 1](https://github.com/demisto/content/pull/11640/files#diff-25a469500140cd4084a2d0cca5db774a5e13c046da6e8bae1eb6aae38432839c) and [RN 2](https://github.com/demisto/content/pull/11640/files#diff-ddbe8ab21df77e49c922d1013d015ba51aff5f0c7aef3cf20977f257927623d3) as the rest are identical.

File: Packs/AzureKubernetesServices/ReleaseNotes/1_0_1.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Azure Kubernetes Services (Beta)
+- Updated **MicrosoftApiModule** to handle ***404 - Page Not Found*** response.
```
Comment (amshamah419) [inline Packs/AzureKubernetesServices/ReleaseNotes/1_0_1.md:4]:
```suggestion
- Updated **MicrosoftApiModule** to handle a ***404 - Page Not Found*** response.
```

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**